src/clr-angular/utils/focus-trap/focus-trap.directive.ts

```diff
@@ -49,7 +49,8 @@ export class FocusTrapDirective implements AfterViewInit, OnDestroy {
   }
 
   private createFocusableOffScreenEl(): any {
-    const offScreenSpan = this.renderer.createElement('span');
+    // Not using Renderer2's createElement method because that leads to DOM leakage.
+    const offScreenSpan = this.document.createElement('span');
```

Comment:
Does this work right with SSR? To be safe, I think we should update `ngAfterViewInit` so it doesn't try run this code branch on the server.

```
    ngAfterViewInit() {
     if (isPlatformBrowser(this.platformId)) {
       this.previousActiveElement = <HTMLElement>this.document.activeElement;
       this.addReboundEls();
     }
   }
```

Reply:
The `addReboundEls()` method itself checks whether it's running in the browser or not. So we only add these off-screen elements when the app is running in a browser.